### PR TITLE
Feature 27 faq system

### PR DESCRIPTION
**Pull request by platonovsimeon**

### CODE REVIEW

File: src/main/java/fi/metatavu/metamind/bot/script/BotFaqServiceScriptBinding.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  * @param baseAPIURL
+  * @return searchResults
+  */
+  public String getAnswers(Value baseAPIURL, Value searchText) {
+    if ( !searchText.isString() || !baseAPIURL.isString() ) {
+      return "";
+    }
+    return faqController.getAnswers(baseAPIURL.asString(), searchText.asString(), null);
+  }
+  public String getAnswers(Value baseAPIURL, Value searchText,Value filterCategoryId) {
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/bot/script/BotFaqServiceScriptBinding.java]: Add documentation

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+import org.json.simple.parser.ParseException;
+import org.slf4j.Logger;
+
+/**
+ * Controller for FAQ-functionality
+ * @author Simeon Platonov
+ *
+ */
+public class FaqController {
+  @Inject
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/faq/FaqController.java]: Add line break

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+   * 
+   * @param searchString
+   * @param category
+   * @param baseAPIURL
+   * @return searchResults
+   * @throws IOException 
+   * @throws ParseException 
+   */
+ 
+  public String getAnswers(String baseAPIURL, String searchText,String filterCategoryId){
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/faq/FaqController.java]: Add documentation

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+   * 
+   * @param url
+   * @return response body in JSON-array format
+   * @throws ParseException
+   * @throws IOException
+   */
+  public JSONArray getResponseJSONArray(URI uri){
+    try {
+      HttpGet httpGet = new HttpGet(uri);
+      CloseableHttpClient httpclient = HttpClients.custom().build();
```
Comment (hkurhinen) [inline src/main/java/fi/metatavu/metamind/faq/FaqController.java]: use try with resources syntax or close httpClient and response

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java (hunk 3 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/faq/FaqController.java]: Updated documentation

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java (hunk 4 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/faq/FaqController.java]: Added .close() statements

File: src/main/java/fi/metatavu/metamind/faq/FaqController.java (hunk 2 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/faq/FaqController.java]: added a line break

File: src/main/java/fi/metatavu/metamind/bot/script/BotFaqServiceScriptBinding.java (hunk 1 above)
Comment (platonovsimeon) [reply src/main/java/fi/metatavu/metamind/bot/script/BotFaqServiceScriptBinding.java]: added documentation

**Pull request merged**